pkg/siteconfig/clusterinstance.go

```diff
@@ -0,0 +1,190 @@
+package siteconfig
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	siteconfigv1alpha1 "github.com/openshift-kni/eco-goinfra/pkg/schemes/siteconfig/v1alpha1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ClusterInstanceBuilder provides struct for the ClusterInstance object.
+type ClusterInstanceBuilder struct {
+	Definition *siteconfigv1alpha1.ClusterInstance
```

Comment:
Please add comments for public elements of the struct

Reply:
@kononovn, I have updated the changes to address your review comments and submitted a new patch for another round of review. Thanks!